virtcontainers/cgroups.go

```diff
@@ -145,45 +145,68 @@ func parentCgroup(hierarchy cgroups.Hierarchy, path string) (cgroups.Cgroup, err
 }
 
 func (s *Sandbox) updateCgroups() error {
+
+	resources, err := s.resources()
+	if err != nil {
+		return err
+	}
+
+	if s.config != nil && s.config.SandboxCgroupOnly {
+		s.Logger().WithField(
+			"SandboxCgroupOnly",
+			s.config.SandboxCgroupOnly).
+			Debug("runtime previously join to sandbox cgroup - nothing to do")
+		return nil
+	}
+
 	if s.state.CgroupPath == "" {
 		s.Logger().Warn("sandbox's cgroup won't be updated: cgroup path is empty")
 		return nil
 	}
 
-	cgroup, err := cgroupsLoadFunc(V1Constraints, cgroups.StaticPath(s.state.CgroupPath))
+	cgroupHierarchy := V1Constraints
+
+	if s.config == nil {
```

Comment:
Can you move this test to the start of the function? It seems a bit odd doing it here as `s.config` doesn't (appear) to be used below this test?!? Moving it up would also allow you to simplify the `s.config.SandboxCgroupOnly` test above.